### PR TITLE
community: Add ZenGuard tool

### PR DESCRIPTION
**Pull request by baur-krykpayev**
** Description**
This is the community integration of ZenGuard AI - the fastest guardrails for GenAI applications. ZenGuard AI protects against:

- Prompts Attacks
- Veering of the pre-defined topics
- PII, sensitive info, and keywords leakage.
- Toxicity
- Etc.

**Twitter Handle** : @zenguardai

- [x] **Add tests and docs**: If you're adding a new integration, please include
  1. Added an integration test
  2. Added colab

- [x] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified.

### CODE REVIEW

**Comment by baur-krykpayev** [comment]
@ccurme et al, was wondering how do I see why the Vercel step is failing. Clicking on the Vercel link is 404 for my account.

File: libs/community/tests/integration_tests/tools/zenguard/test_zenguard.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import pytest
+
+from langchain_community.tools.zenguard.tools import Detector, ZenGuardTool
+
+
+@pytest.fixture()
+def zenguard_tool() -> ZenGuardTool:
+    zenguard_api_key = os.getenv("ZENGUARD_API_KEY")
+    if zenguard_api_key is None:
+        zenguard_api_key = "3Ev_DGvELv7EnlgWMTlpmWTo82tpstyz4Li_R7kTDQw"
```
Comment (ccurme) [inline libs/community/tests/integration_tests/tools/zenguard/test_zenguard.py:13]:
```suggestion
```
It looks like maintainers aren't allowed to modify this branch but I'd remove this and consider the API key compromised.

Comment (baur-krykpayev) [reply libs/community/tests/integration_tests/tools/zenguard/test_zenguard.py]: yup, this API key is blacklisted from the very beginning :)

**Comment by baur-krykpayev** [comment]
ping ping @ccurme

**Comment by baur-krykpayev** [comment]
@ccurme ping ping

**Comment by yaksh0nti** [comment]
close dummy-AI-inc/dummy-ai#765

**Comment by baur-krykpayev** [comment]
@ccurme ping ping for the review

**Comment by baur-krykpayev** [comment]
@baskaryan @eyurtsev please review 🙏

File: .github/workflows/_integration_test.yml [hunk 2]
```diff
@@ -85,6 +85,7 @@ jobs:
           VOYAGE_API_KEY: ${{ secrets.VOYAGE_API_KEY }}
           COHERE_API_KEY: ${{ secrets.COHERE_API_KEY }}
           UPSTAGE_API_KEY: ${{ secrets.UPSTAGE_API_KEY }}
+          ZENGUARD_API_KEY: ${{ secrets.ZENGUARD_API_KEY }}
```
Comment (ccurme) [inline .github/workflows/_integration_test.yml:88]:
```suggestion
```

**Review by ccurme — APPROVED** [review]
LGTM but heads up that langchain CI will not run integration tests for this.

File: .github/workflows/_integration_test.yml (hunk 2 above)
Comment (baur-krykpayev) [reply .github/workflows/_integration_test.yml]: done. removed.

**Pull request merged**